setup.py

```diff
@@ -13,9 +13,14 @@
 # limitations under the License.
 
 """Setup.py script for the Advanced Reactor Modeling Interface (ARMI)"""
-import re
 from setuptools import setup, find_packages
+import os
+import pathlib
+import re
 
+# grab __version__ from meta.py, without calling __init__.py
+this_file = pathlib.Path(__file__).parent.absolute()
+exec(open(os.path.join(this_file, "armi", "meta.py"), "r").read())
```

Comment:
maybe worth putting a comment in meta.py saying that each line will be executed line by line during setup.py? probably worth pointing to this at least since it may be hard to find/surprising.

Better to just read the version out of the code with a RE?

Reply:
1) Sure, I like the idea of that comment. Though, to be fair, not more code will be "executed" in `meta.py` than if we just did "from armi import meta". That's the only level of execution abound.  I'm just trying to save us from executing everything in `armi/__init__.py` by doing "from armi import __version__", which would also work.
2) We could do regex for this. I have done things like that in the past. But it's always problematic. I prefer to use Python to read Python than to write our own parser to read Python, you know?